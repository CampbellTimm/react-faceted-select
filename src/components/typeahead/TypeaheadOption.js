import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export default class TypeaheadOption extends React.Component {
    static propTypes = {
        classes: PropTypes.object,
        result: PropTypes.string,
        onClickOption: PropTypes.func,
        children: PropTypes.string,
        hover: PropTypes.bool
    };

    static defaultProps = {
        classes: {},
        onClickOption: (event) => event.preventDefault()
    };

    getClasses = () => {
        const classes = {
            'typeahead-option': true,
        };
        classes[this.props.classes.listAnchor] = !!this.props.classes.listAnchor;
        return classNames(classes);
    };

    onClickOption = (event) => {
        event.preventDefault();
        return this.props.onClickOption(this.props.result);
    };

    render() {
        const classes = {
            hover: this.props.hover,
        };
        classes[this.props.classes.listItem] = !!this.props.classes.listItem;

        return (
            <li className={classNames(classes)}>
                <a
                    href="#"
                    id="typeahead-option"
                    onClick={this.onClickOption}
                    className={this.getClasses()}
                    ref="anchor"
                >
                    {this.props.children}
                </a>
            </li>
        );
    }
}
