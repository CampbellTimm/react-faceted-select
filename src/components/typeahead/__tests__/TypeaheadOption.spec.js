import React from 'react';
import { shallow } from 'enzyme';
import TypeaheadOption from '../TypeaheadOption';

describe('TypeaheadOption', () => {

    it('should render TypeaheadOption', () => {
        const onClickOption = jest.fn();
        const preventDefault = jest.fn();

        const component = shallow(
            <TypeaheadOption
                classes={{}}
                result={"result"}
                onClickOption={onClickOption}
            >
                John Doe
            </TypeaheadOption>
        );

        component
            .find('#typeahead-option')
            .simulate('click', { preventDefault });

        expect(onClickOption).toHaveBeenCalled();
        expect(onClickOption.mock.calls[0][0]).toEqual("result");
        expect(component).toMatchSnapshot();
    });

});