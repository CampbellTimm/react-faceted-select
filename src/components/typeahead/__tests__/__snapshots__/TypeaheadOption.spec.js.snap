// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeaheadOption should render TypeaheadOption 1`] = `
<li
  className=""
>
  <a
    className="typeahead-option"
    href="#"
    id="typeahead-option"
    onClick={[Function]}
  >
    John Doe
  </a>
</li>
`;
